src/pages/TermsOfService.ts

```diff
@@ -0,0 +1,148 @@
+import _ from 'lodash/fp';
+import { useState } from 'react';
+import { div, h, h1, img } from 'react-hyperscript-helpers';
+import { ButtonOutline, ButtonPrimary, ButtonSecondary } from 'src/components/common';
+import { centeredSpinner } from 'src/components/icons';
+import { MarkdownViewer, newWindowLinkRenderer } from 'src/components/markdown';
+import scienceBackground from 'src/images/science-background.jpg';
+import { Ajax } from 'src/libs/ajax';
+import { signOut } from 'src/libs/auth';
+import colors from 'src/libs/colors';
+import { reportError, withErrorReporting } from 'src/libs/error';
+import * as Nav from 'src/libs/nav';
+import { useOnMount } from 'src/libs/react-utils';
+import { authStore } from 'src/libs/state';
+import * as Style from 'src/libs/style';
+import * as Utils from 'src/libs/utils';
+
+const TermsOfServicePage = () => {
+  const [busy, setBusy] = useState<boolean>();
+  const { signInStatus, termsOfService } = authStore.get(); // can't change while viewing this without causing it to unmount, so doesn't need to subscribe
+  const acceptedLatestTos = signInStatus === 'signedIn' && termsOfService.userHasAcceptedLatestTos;
+  const usageAllowed = signInStatus === 'signedIn' && termsOfService.permitsSystemUsage;
+  const [tosText, setTosText] = useState<string>();
+
+  useOnMount(() => {
+    const loadTosAndUpdateState = _.flow(
+      Utils.withBusyState(setBusy),
+      withErrorReporting('There was an error retrieving our terms of service.')
+    )(async () => {
+      setTosText(await Ajax().User.getTos());
+    });
+    loadTosAndUpdateState();
+  });
+
+  const accept = async () => {
+    try {
+      setBusy(true);
+      const acceptTosResponse = await Ajax().User.acceptTos();
+      const { enabled } = acceptTosResponse || { enabled: false };
+      if (enabled) {
+        const termsOfServiceResponse = await Ajax().User.getTermsOfServiceComplianceStatus();
+        const termsOfService = termsOfServiceResponse || { userHasAcceptedLatestTos: false, permitsSystemUsage: false };
+
+        const registrationStatus = 'registered';
+        authStore.update((state) => ({ ...state, registrationStatus, termsOfService }));
+        Nav.goToPath('root');
+      } else {
+        reportError('Error accepting TOS, unexpected backend error occurred.', new Error('Unexpected backend error'));
```

Comment:
"Terms of service" is spelled out elsewhere. Capitalization seems to be inconsistent though.

```suggestion
        reportError('Error accepting terms of service, unexpected backend error occurred.', new Error('Unexpected backend error'));
```